services/QuillLMS/app/models/subscription.rb

```diff
@@ -125,6 +142,42 @@ class RenewalNilStripeCustomer < StandardError; end
   scope :started, -> { where("start_date <= ?", Date.current) }
   scope :paid_with_card, -> { where.not(stripe_invoice_id: nil).or(where(payment_method: 'Credit Card')) }
 
+  def self.create_with_subscriber_join(subscriber, subscription_attrs)
+    if !subscription_attrs[:expiration]
+      case subscription_attrs[:account_type]
+      when TEACHER_TRIAL
+        subscription_attrs = subscription_attrs.merge(Subscription.set_trial_expiration_and_start_date(subscriber))
+      when CB_LIFETIME_SUBSCRIPTION_TYPE
+        subscription_attrs = subscription_attrs.merge(Subscription.set_cb_lifetime_expiration_and_start_date)
+      else
+        subscription_attrs = subscription_attrs.merge(Subscription.set_premium_expiration_and_start_date(subscriber))
+      end
+    end
+
+    subscriber.subscriptions.each do |existing_subscription|
+      existing_subscription.update!(recurring: false)
+      existing_subscription.stripe_cancel_at_period_end
+    end
+
+    subscription = Subscription.create!(subscription_attrs)
+    subscriber.join_subscription(subscription)
+    subscription
+  end
+
+  def audience
+    return DISTRICT if district_subscriptions.exists?
+    return SCHOOL if school_subscriptions.exists?
+    return TEACHER if user_subscriptions.exists?
+  end
+
+  def premium_types
```

Comment:
Just noting that this doesn't have a default, so could return `nil`. That might be unexpected downstream. Might be better to default to `[]` at least.